administration/src/bp-modules/applications/NoteDialogController.tsx

```diff
@@ -0,0 +1,83 @@
+import { Button, Tooltip } from '@blueprintjs/core'
+import React, { ReactElement } from 'react'
+import styled from 'styled-components'
+
+import getMessageFromApolloError from '../../errors/getMessageFromApolloError'
+import { useUpdateApplicationNoteMutation } from '../../generated/graphql'
+import { useAppToaster } from '../AppToaster'
+import TextAreaDialog from '../components/TextAreaDialog'
+import { Application } from './ApplicationsOverview'
+
+const NoteButton = styled(Button)`
+  margin-right: 10px;
+  align-self: flex-start;
+  @media print {
+    display: none;
+  }
+`
+
+const MultilineContent = styled(Tooltip)`
+  white-space: pre-wrap;
+`
+
+type NoteDialogControllerProps = {
+  application: Application
+  onOpenNoteDialog: (value: boolean) => void
+  isOpen: boolean
+}
+
+const EXCERPT_LENGTH = 80
+
+const NoteDialogController = ({
+  application,
+  onOpenNoteDialog,
+  isOpen,
+}: NoteDialogControllerProps): ReactElement | null => {
+  const appToaster = useAppToaster()
+  const [updateApplicationNote, { loading, data: applicationData }] = useUpdateApplicationNoteMutation({
+    onError: error => {
+      const { title } = getMessageFromApolloError(error)
+      appToaster?.show({ intent: 'danger', message: title })
+    },
+    onCompleted: () => {
+      appToaster?.show({ intent: 'success', message: 'Notiz erfolgreich geändert.', timeout: 2000 })
+    },
+  })
+
+  const onClose = () => onOpenNoteDialog(false)
+
+  const onSave = (text: string) => {
+    updateApplicationNote({ variables: { applicationId: application.id, text } })
+    onOpenNoteDialog(false)
+  }
+
+  const getNoteExcerpt = (maxChars: number, text: string): string =>
+    text.length > maxChars ? `${text.slice(0, maxChars)} ...` : text
+
+  const note = applicationData ? applicationData.applicationNote : application.note
```

Comment:
So this is used to show only the updated note if the note was updated, right? :D

Hmm.. can't we update the original `application` object similarly to how we remove an application object on delete?